python/dolfin/io.py

```diff
@@ -118,7 +118,7 @@ class XDMFFile:
     # Import encoding (find better way?)
     Encoding = cpp.io.XDMFFile.Encoding
 
-    def __init__(self, mpi_comm, filename: str):
+    def __init__(self, mpi_comm, filename: str, encoding=Encoding.HDF5):
```

Comment:
Having a default doesn't make sense when reading.

Reply:
Agree. Later with file modes we can make overloaded constructor that takes file mode and encoding and throws on "r" mode. 
In python layer we can default encoding=None and dispatch inside __init__.py. Makes sense?